aci/resource_aci_cloudtemplateregiondetail.go

```diff
@@ -0,0 +1,199 @@
+			"parent_dn": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"hub_networking": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"disable",
+					"enable",
+				}, false),
+			},
+		})),
+	}
+}
+
+func getActualHubNetworkingValue(hub_networking string) string {
+	if hub_networking == "disable" {
+		return "no"
+	} else {
+		return "yes"
+	}
+}
+
+func setHubNetworkingValue(hub_networking string) string {
+	if hub_networking == "no" {
+		return "disable"
+	} else {
+		return "enable"
+	}
+}
+
+func getRemoteCloudTemplateRegion(client *client.Client, dn string) (*models.CloudTemplateRegion, error) {
+	cloudtemplateRegionDetailCont, err := client.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+	cloudtemplateRegionDetail := models.CloudTemplateRegionFromContainer(cloudtemplateRegionDetailCont)
+	if cloudtemplateRegionDetail.DistinguishedName == "" {
+		return nil, fmt.Errorf("Cloud Template Region %s not found", dn)
+	}
+	return cloudtemplateRegionDetail, nil
+}
+
+func setCloudTemplateRegionAttributes(cloudtemplateRegionDetail *models.CloudTemplateRegion, d *schema.ResourceData) (*schema.ResourceData, error) {
+	d.SetId(cloudtemplateRegionDetail.DistinguishedName)
+	cloudtemplateRegionDetailMap, err := cloudtemplateRegionDetail.ToMap()
+	if err != nil {
+		return d, err
+	}
+	dn := d.Id()
+	if dn != cloudtemplateRegionDetail.DistinguishedName {
+		d.Set("parent_dn", "")
+	} else {
+		d.Set("parent_dn", GetParentDn(cloudtemplateRegionDetail.DistinguishedName, fmt.Sprintf("/"+models.RnCloudtemplateRegionDetail)))
+	}
+	d.Set("annotation", cloudtemplateRegionDetailMap["annotation"])
+	d.Set("hub_networking", setHubNetworkingValue(cloudtemplateRegionDetailMap["hubNetworkingEnabled"]))
+	return d, nil
+}
+
+func resourceAciCloudTemplateRegionImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+	cloudtemplateRegionDetail, err := getRemoteCloudTemplateRegion(aciClient, dn)
+	if err != nil {
+		return nil, err
+	}
+	schemaFilled, err := setCloudTemplateRegionAttributes(cloudtemplateRegionDetail, d)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("[DEBUG] %s: Import finished successfully", d.Id())
+	return []*schema.ResourceData{schemaFilled}, nil
+}
+
+func resourceAciCloudTemplateRegionCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	log.Printf("[DEBUG] CloudTemplateRegion: Beginning Creation")
+	aciClient := m.(*client.Client)
+	CloudProviderandRegionNamesDn := d.Get("parent_dn").(string)
+
+	cloudtemplateRegionDetailAttr := models.CloudTemplateRegionAttributes{}
+
+	if Annotation, ok := d.GetOk("annotation"); ok {
+		cloudtemplateRegionDetailAttr.Annotation = Annotation.(string)
+	} else {
+		cloudtemplateRegionDetailAttr.Annotation = "{}"
```

Comment:
I don't think APIC accepts "" for annotation. 

We use a conversion in the models file like this......

```
	if value == "{}" {
		data[key] = ""
	}
```

Reply:
I think we enforce annotation to have an empty value in the form of {} and in turn assure its existence in the payload because of the omitempty flag in the json struct for the annotation in the model files. Annotation would not be included in the payload if it were just ""